src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
> Theoretically one could want to extend/decorate the ChainDecoder specifically and in that case you may need the concrete class implementation details

In my opinion, it indicates a design issue: something like `DecoderResolver` is missing.

Also, I don't understand what's the point of specifying runtime exceptions in the `@throws` tag. By analogy with `unchecked exceptions` from Java, these exceptions are mostly for an errors: no decoder found, bad configuration, bug, etc. User of an interface doesn't catch them, because he can't "fix" it. He has to fix the code, configuration, etc. in order to resolve his problem.